02-peer-protocol.md

```diff
@@ -215,65 +219,96 @@ acceptance of the new channel.
    * [`33`:`payment_basepoint`]
    * [`33`:`delayed_payment_basepoint`]
    * [`33`:`first_per_commitment_point`]
+   * [`2`:`len`]
+   * [`len`:`funding_tx`]
+   * [`64`:`signature`]
 
 #### Requirements
 
-
-The receiving MUST reject the channel if the `chain_hash` value within the
+The receiving node MUST reject the channel if the `chain_hash` value within the
 `open_channel` message is set to a hash of a chain unknown to the receiver.
 
 The `temporary_channel_id` MUST be the same as the `temporary_channel_id` in the `open_channel` message.  The sender SHOULD set `minimum_depth` to a number of blocks it considers reasonable to avoid double-spending of the funding transaction.
 
 The receiver MAY reject the `minimum_depth` if it considers it unreasonably large.
 Other fields have the same requirements as their counterparts in `open_channel`.
 
+The sender can choose to also fund the channel, so that they can start with a positive balance and can use the channel to pay right away.
+
+The sender MUST set `len` to the length of an unsigned funding transaction, followed by the serialized funding transaction.
+`funding_tx` is an unsigned funding transaction which includes:
+ * optional additional inputs that the sender wants to use to fund the channel. These inputs MUST be either P2WPKH or P2WSH and must include the witness script in their signature script field.
+ * an optional change output
+ * a funding transaction output as defined in [BOLT #3](03-transactions.md#funding-transaction-output)
+The sender MUST then set `signature` to the valid signature using its funding_pubkey for the initial commitment transaction as defined in BOLT #3
+ 
+The amount that the sender will start with, i.e. that will be included in his first commit transaction, is the difference between the additional inputs
+and the change output.
+
+If the sender does not wish to fund the channel, they MAY set `len` to 0 and not include the funding transaction since the
```

Comment:
yes having the spend the funding outputs when channel opening times out is annoying. I think that it can be mitigated by adding time-outs to the funding output scripts, I'll post an update asap